src/main/java/Airport.java

```diff
@@ -1,55 +1,79 @@
 
 import java.util.*;
 
+import static com.sun.xml.internal.ws.spi.db.BindingContextFactory.LOGGER;
+
 
 public class Airport {
     private List<? extends Plane> planes;
 
 
     public List<PassengerPlane> getPassengersPlane() {
-        List<? extends Plane> planesList = this.planes;
         List<PassengerPlane> passengersPlaneList = new ArrayList<>();
-        for (Plane plane : planesList) {
-            if (plane instanceof PassengerPlane) {
-                passengersPlaneList.add((PassengerPlane) plane);
-            }
+        for (Plane plane : planes) {
+            addPlaneIfItIsPassenger(plane, passengersPlaneList);
         }
         return passengersPlaneList;
     }
 
+    public List<PassengerPlane> addPlaneIfItIsPassenger(Plane plane, List<PassengerPlane> passengerPlaneList) {
+        if (plane instanceof PassengerPlane) {
+            passengerPlaneList.add((PassengerPlane) plane);
+        } else {
+            LOGGER.info("Plane is not passenger");
+        }
+        return passengerPlaneList;
+    }
+
+
     public List<MilitaryPlane> getMilitaryPlanes() throws NonMilitaryPlainException {
         List<MilitaryPlane> militaryPlanes = new ArrayList<MilitaryPlane>();
         for (Plane plane : planes) {
-            if (plane instanceof MilitaryPlane) {
-                militaryPlanes.add((MilitaryPlane) plane);
-            } else {
-                throw new NonMilitaryPlainException("Plane is not military");
-            }
+            addPlaneIfItIsMilitary(plane, militaryPlanes);
         }
         return militaryPlanes;
     }
 
-    public PassengerPlane getPassengerPlaneWithMaxPassengersCapacity() {
+    public List<MilitaryPlane> addPlaneIfItIsMilitary(Plane plane, List<MilitaryPlane> militaryPlanes) {
+        if (plane instanceof MilitaryPlane) {
+            militaryPlanes.add((MilitaryPlane) plane);
+        } else {
+            LOGGER.info("Plane is not military");
+        }
+        return militaryPlanes;
+    }
+
+    public PassengerPlane getPassengerPlaneWithMaxPassengersCapacity() throws NonMilitaryPlainException {
         List<PassengerPlane> passengerPlanes = getPassengersPlane();
         PassengerPlane planeWithMaxCapacity = passengerPlanes.get(0);
-        for (int i = 0; i < passengerPlanes.size(); i++) {
+        return findInCyclePlaneWithMaxCapacity(passengerPlanes, planeWithMaxCapacity);
+    }
+
+    public PassengerPlane findInCyclePlaneWithMaxCapacity(List<PassengerPlane> passengerPlanes, PassengerPlane planeWithMaxCapacity) throws NonMilitaryPlainException {
+        for (int i = 0; i < getPassengersPlane().size(); i++) {
             if (passengerPlanes.get(i).getPassengersCapacity() > planeWithMaxCapacity.getPassengersCapacity()) {
                 planeWithMaxCapacity = passengerPlanes.get(i);
             }
         }
         return planeWithMaxCapacity;
     }
 
+
     public List<MilitaryPlane> getTransportMilitaryPlanes() throws NonMilitaryPlainException {
```

Comment:
firstly, like in comments above. you need to get rid of iterating same cycles over 2 method and make iteration in 1, logic in the second.

Reply:
second,
do you see that it's the same logic in findInCycleBomberMilitaryPlanes and findInCycleMilitaryPlane ?
try to add only 1 method performing this logic.
ex:

public List<MilitaryPlane> getMilitaryPlanesByType(MilitaryType type) throws NonMilitaryPlainException {
List<MilitaryPlane> militaryPlanesByType = new ArrayList<>();
for (int i = 0; i < militaryPlanes.size(); i++) {
MilitaryPlane militaryPlain = militaryPlanes.get(i);
if (isExpectedMilitaryPlaneType(militaryPlain, type){
militaryPlanesByType.add(militaryPlain);
}
return militaryPlanesByType;
}

boolean isExpectedMilitaryPlaneType(MilitaryPlane plane, MilitaryType exp){
return plane.getType() == MilitaryType ? true : false;
}